andela/odin-ah-backend
# #161285613 Use Sequelize API in place of SQL queries

#### What does this PR do?
- Uses Sequelize APIs to implement article search functionality

#### Description of Task to be completed?
- Refactor search controller to use Sequelize APIs
- Create SearchEngine class for search extensibility

#### How should this be manually tested?
- Fetch and checkout to the branch
```$xslt
> git fetch
> git checkout bg-161285613-remove-raw-sql-from-search-feature
> npm install 
```
- Setup database using .env-sample
* Run Sequelize migration 
```$xslt
 > npm run migration
 ```
- Run application
```$xslt 
> npm run start
```
- To run test
```$xslt 
> npm run test
```

* Create some articles and add tags using the article endpoint. See #24 for details

* Using a REST client such as postman to test endpoints: 
To search for an article: `GET /api/v1/search?q=<query>&limit=<limit>&page=<page>&author=<author>&tag=<tag>` 

* Replace the values in the angle brackets with appropriate inputs. See the API documentation at `http://localhost:3000/api-docs/` for details

* Sample request:
```json
GET /api/v1/search?q=train+dragon&limit=10&page=1&author=3
```

#### What are the relevant pivotal tracker stories?
- [#161285613](https://www.pivotaltracker.com/story/show/161285613)
